internal/moduletest/assertion.go

```diff
@@ -0,0 +1,80 @@
+package moduletest
+
+import (
+	"github.com/hashicorp/terraform/tfdiags"
+)
+
+// Assertion is the description of a single test assertion, whether
+// successful or unsuccessful.
+type Assertion struct {
+	Outcome Status
+
+	// Description is a user-provided, human-readable description of what
+	// this assertion represents.
+	Description string
+
+	// Message is typically relevant only for TestFailed or TestError
+	// assertions, giving a human-readable description of the problem,
+	// formatted in the way our format package expects to receive paragraphs
+	// for terminal word wrapping.
+	Message string
+
+	// Diagnostics includes diagnostics specific to the current test assertion,
+	// if available.
+	Diagnostics tfdiags.Diagnostics
+}
+
+// Component represents a component being tested, each of which can have
+// several associated test assertions.
+type Component struct {
+	Assertions map[string]*Assertion
+}
+
+// AssertionKey is a unique identifier for an Assertion within a particular
+// test run.
+type AssertionKey struct {
```

Comment:
I can't find any references to this type yet. What's it for?

Reply:
Oh, good catch! This was from an earlier iteration of the model where I had the component just be a field associated with each assertion rather than its own distinct level of hierarchy (matching the JUnit XML view of things) but then I refactored it into a three-level hierarchy later and forgot to delete this unused type.

I'll delete it now.